File: backend/models/Spot.js
// models/Spot.js

import mongoose from 'mongoose'

const spotSchema = new mongoose.Schema({

    title: {
        type: String,
        required: true,
    },

    description: String,

    location: {
        type: {
          type: String,
          enum: ['Point'],
          default: 'Point',
        },
        coordinates: {
          type: [Number],
          required: true,
        },
    },

    images: [String],

    tags: [String],

    rating: {
        type: Number,
        default: 0,
    },

    createdAt: {
        type: Date,
        default: Date.now,
    },

    createdBy: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User', // Referenz auf das User-Modell
        required: true,
    },

    comments: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'Comment'
        }
    ],

    upvotes: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'User'
        }
    ],

    downvotes: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'User'
        }
    ]

})

spotSchema.index({ location: '2dsphere' })

const Spot = mongoose.model('Spot', spotSchema)
export default Spot

File: frontend/lib/models/spot.dart
class Spot {
  final String id;
  final String title;
  final String description;
  final double latitude;
  final double longitude;

  Spot({
    required this.id,
    required this.title,
    required this.description,
    required this.latitude,
    required this.longitude,
  });

  factory Spot.fromJson(Map<String, dynamic> json) {

    final coords = json['location']['coordinates'];

    return Spot(
      id: json['_id'],
      title: json['title'],
      description: json['description'],
      longitude: coords[0],
      latitude: coords[1],
    );

  }

}